book/src/leader-rotation.md

```diff
@@ -3,17 +3,46 @@
 At any given moment, a cluster expects only one fullnode to produce ledger
 entries. By having only one leader at a time, all validators are able to replay
 identical copies of the ledger. The drawback of only one leader at a time,
-however, is that a malicious leader is cabable of censoring votes and
+however, is that a malicious leader is capable of censoring votes and
 transactions. Since censoring cannot be distinguished from the network dropping
 packets, the cluster cannot simply elect a single node to hold the leader role
-indefinitely. Instead, the cluster minimizes the influence of a malcioius
+indefinitely. Instead, the cluster minimizes the influence of a malicious
 leader by rotating which node takes the lead.
 
 Each validator selects the expected leader using the same algorithm, described
 below. When the validator receives a new signed ledger entry, it can be certain
 that entry was produced by the expected leader.
 
-## Leader Schedule Generation
+## Leader Schedule Rotation
+
+Leader schedule is generated every epoch when the root fork crosses the
+epoch boundary.
+
+1. The root fork is updated as a validator votes for new forks.
+
+2. When the root fork slot height crosses the epoch boundary, the leader
+schedule is updated for the next epoch.
+
+For example:
+
+The epoch is 100 slots. The root fork is updated from fork computed at slot 99
+to a fork computed at slot 102. Slot 100,101 forks were skipped because of
+failures.  The new leader schedule is computed using fork 102.  It is active
```

Comment:
this may introduce stake skews, differing leader schedules...

Reply:
I guess stake skews are unavoidable unless the network restricts when stakes may be placed.

Should this document include information about how many epochs in advance of schedule generation a stake must be placed before it is considered for leader rotation?

Reply:
@rob-solana 
Stakes are valid only in the subsequent leader schedule

Reply:
@rob-solana 

I dont see how a skew is possible.  Everyone will observe the same root fork, 102, because if they are following locktower correctly there is no way for any other root to occur without network wide commitment.  So any new stakes present in 102 will become part of the schedule for 200+.

Reply:
no skews can exist because *everyone* has skipped 100 and 101 when their root reaches 102